spark/src/main/java/org/apache/iceberg/spark/SparkReadOptions.java

```diff
@@ -42,6 +42,9 @@ private SparkReadOptions() {
   // Overrides the table's read.split.open-file-cost
   public static final String FILE_OPEN_COST = "file-open-cost";
 
+  // Overrides the table's read.split.by-partition
+  public static final String BY_PARTITION = "by-partition";
```

Comment:
Is this useful for a user-facing setting? I don't know when a user would want to set this by hand.